clients/src/main/java/org/apache/kafka/common/security/scram/internals/ScramMechanism.java

```diff
@@ -23,13 +23,15 @@
 
 public enum ScramMechanism {
 
-    SCRAM_SHA_256("SHA-256", "HmacSHA256", 4096),
-    SCRAM_SHA_512("SHA-512", "HmacSHA512", 4096);
+    SCRAM_SHA_256((byte) 1, "SHA-256", "HmacSHA256", 4096, 16384),
```

Comment:
This duplicates the type code that is found in `org.apache.kafka.clients.admin.ScramMechanism`. We need to make sure that these don't get out of sync, and also that they're not changed, because they are part of the binary ABI.

Please add a comment referring to the other enum here. Also add a comment in `org.apache.kafka.clients.admin.ScramMechanism` referring to this enum. Make sure to mention that changing the type codes would break backwards compatibility.

Reply:
Yes, and I'd like to consolidate them so that we only have one in the future, but this isn't the commit that should do that. I'll add a comment and I understand the the implications that it is a binary ABI.

Reply:
Comment added!